### ISSUE TITLE
cannot use for google login

### ISSUE THREAD

**Issue by mayah** [alexcrichton/oauth2-rs#10]
Hi, I tried this crate a bit for google login. Google requires `response_type` in authorization url, but there is no way to do that. As mentioned in #8 a bit, we need to have `response_type`.

**Comment by lipanski** [comment]
hi @mayah 

you can still set the `response_type` even with the current implementation:

```rust
let mut authorize_url = config.authorize_url(String::new());
authorize_url.query_pairs_mut().append_pair("response_type", "code");
```

this should happen somewhere before you call `config.exchange_code(code)`.

my issue #8 is about improving the interface, but after https://github.com/alexcrichton/oauth2-rs/pull/7 I got the Google flow working.

**Comment by lipanski** [comment]
here's the entire code (in this case for Google Calendar):

```rust
    let mut config = Oauth2Config::new(
        "YOUR_CLIENT_ID",
        "YOUR_CLIENT_SECRET",
        "https://accounts.google.com/o/oauth2/v2/auth",
        "https://www.googleapis.com/oauth2/v3/token"
    );

    // Point to your redirect URL
    config.redirect_url = "http://localhost:8080".to_string();

    config.scopes = Vec::new();
    config.scopes.push("https://www.googleapis.com/auth/calendar".to_string());

    let mut authorize_url = config.authorize_url(String::new());
    authorize_url.query_pairs_mut().append_pair("response_type", "code");

    let mut code = String::new();

    // This is my own-the-fly redirect URL server: Ignore this part if you've got this handled somewhere else
    let listener = TcpListener::bind("127.0.0.1:8080").unwrap();
    for stream in listener.incoming() {
        match stream {
            Ok(stream) => {
                let mut reader = BufReader::new(stream);

                let mut request_line = String::new();
                reader.read_line(&mut request_line).unwrap();

                let redirect_url = request_line.split_whitespace().nth(1).unwrap();
                let url = Url::parse(&("http://localhost".to_string() + redirect_url)).unwrap();

                let code_pair = url.query_pairs().find(|pair| {
                    let &(ref key, _) = pair;
                    key == "code"
                }).unwrap();

                let (_, value) = code_pair;
                code = value.into_owned();

                break;
            }
            Err(e) => {}
        }
    };

    let token = config.exchange_code(code).unwrap();
```

**Comment by mayah** [comment]
yeah, I'm ok with adding query params after taking authorize_url, however, I don't think it's intuitive. If it's ok, then it might be also ok if we construct authorize_url without this kind of library, right?

Anyway, it looks you're working on enhancing the api interface, I'll wait for your work. (or, I'm ok to make a PR to add an interface.)

**Comment by lipanski** [comment]
@mayah I've closed that monster ticket and just released 1.0.0 => have a look if it's still of interest. the documentation is available [here](https://docs.rs/oauth2/1.0.0/oauth2/).

**Comment by mayah** [comment]
Hi, thanks for the info! I'll check it.
